### ISSUE TITLE
`clippy::reversed_empty_ranges` does not warn for ranges x..y where x == y

### ISSUE THREAD

**Issue by harudagondi** [rust-lang/rust-clippy#10167]
### Summary

According to the [description](https://rust-lang.github.io/rust-clippy/master/index.html#reversed_empty_ranges):

> ## What it does
>
> Checks for range expressions x..y where both x and y are constant and x is greater or equal to y.

However, this do not warn for ranges where x is equal to y.

### Lint Name

reversed_empty_ranges

### Reproducer

I tried this code:

```rust
let x = [1, 2, 3, 4, 5, 6, 7];
dbg!(&x[5..1]);
dbg!(&x[5..5]);
```

I expected to see this happen: Warnings for both `dbg!` calls.

Instead, this happened: Only the first one emits a warning.

```
    Checking playground v0.0.1 (/playground)
error: this range is reversed and using it to index a slice will panic at run-time
 --> src/main.rs:3:13
  |
3 |     dbg!(&x[5..1]);
  |             ^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#reversed_empty_ranges
  = note: `#[deny(clippy::reversed_empty_ranges)]` on by default

error: could not compile `playground` due to previous error
```

### Version

```text
rustc 1.66.0 (69f9c33d7 2022-12-12)
binary: rustc
commit-hash: 69f9c33d71c871fc16ac445211281c6e7a340943
commit-date: 2022-12-12
host: x86_64-pc-windows-msvc
release: 1.66.0
LLVM version: 15.0.2

and

rustc 1.68.0-nightly (0468a00ae 2022-12-17)
binary: rustc
commit-hash: 0468a00ae3fd6ef1a6a0f9eaf637d7aa9e604acc
commit-date: 2022-12-17
host: x86_64-pc-windows-msvc
release: 1.68.0-nightly
LLVM version: 15.0.6
```

**Comment by blyxyas** [comment]
Some functions can accept `N..N` as a valid range, that's why this lint was limited to only lint in that case if the range was being used in a `for` loop. Discussed and explained by @ ebroto in #5689 and #5628.

For example, while `dbg!(&s[5..5]);` doesn't really have useful results, `v.splice(0..0, std::iter::repeat(0).take(4));` changes a vector in a useful (adding 0 zeros for the first index).

As Clippy cannot know if a `N..N` is accepted in particular cases, the lint will probably continue as is.
I will unnasign myself due to this.

@rustbot release-assignment

**Comment by harudagondi** [comment]
Yeah, N..N makes sense to me. I would rather reword the doc to say "x is greater than y", rather than changing the current clippy behavior.
